kalite/control_panel/tests/control_panel.py

```diff
@@ -11,6 +11,51 @@
 logging = settings.LOG
 
 
+class DeviceRegistrationTests(FacilityMixins,
+                       StudentProgressMixin,
+                       BrowserActionMixins,
+                       CreateZoneMixin,
+                       CreateAdminMixin,
+                       KALiteBrowserTestCase):
+
+    def setUp(self):
+        self.admin_data = {"username": "admin", "password": "admin"}
+        self.admin = self.create_admin(**self.admin_data)
+
+        super(DeviceRegistrationTests, self).setUp()
+
+
+    def test_device_registration_availability(self):
+        """
+        This test simulate the device registration availability.
+        The Registration button must appear else the test will fail.
+        """
+        facility_name = 'default'
+        self.fac = self.create_facility(name=facility_name)
+        self.browser_login_admin(**self.admin_data)
+        self.browse_to(self.reverse('zone_redirect'))  # zone_redirect so it will bring us to the right zone
+
+        device_status = self.browser.find_elements_by_class_name("not-registered-only")
+        expected_result = "Register Now!"
+        self.assertEqual(device_status[0].text, expected_result)
```

Comment:
Looks like both elements (meaning those with class `not-registered-only` and `registered-only` are included in the page, it's just the visibility that's set depending on the registration status. So I'd expect the assertions in this test and the next one to be true independent of whether the device is registered or not, e.g. the element will be in the DOM and will have that text, it just won't be visible. Further, my experience so far is leading me to believe that `find` methods in Selenium without an implicit or explicit wait can cause trouble.

Therefore, I suggest using [WebDriverWait.until]() with the [visibilit_of](http://selenium-python.readthedocs.org/api.html#selenium.webdriver.support.expected_conditions.visibility_of) expected condition, along with a `try` block, something like this:

```python
try:
    WebDriverWait(self.browser, some_timeout).until(EC.visibility_of(some_element))
except TimeoutException:    # until will throw this exception if the element is not visible
    # Fail the test
# Otherwise pass the test.
```